sdk/template/azure-template/test/CMakeLists.txt

```diff
@@ -9,16 +9,18 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 add_compile_definitions(AZURE_TEST_DATA_PATH="${CMAKE_BINARY_DIR}")
 
+include(GoogleTest)
+
 add_executable (
      azure-template-test
      ut/template_test.cpp
      )
 
-target_link_libraries(azure-template-test PRIVATE Azure::azure-template)
+target_link_libraries(azure-template-test PRIVATE Azure::azure-template gtest_main)
```

Comment:
Why do we use `gtest_main` here but `gtest` and `gmock` in azure-core above?

Reply:
`gtest_main` adds a `main()` function so we don't need to create the tests starting point. We just need to define the tests we want. This is convenient when we don't need to do any specific stuff before starting the tests.

If we require some specific calls for the entire application and before and afters tests run, we need to link with gets only and define our main() function.
For Core, we need to call curl global init and global clean up methods per application. That's why you see this difference.

Yes, it might be good to mention this with code comments! hahha (starting to read your mind :)